File: src/pewmode/Fires.java
package pewmode;

import org.newdawn.slick.SlickException;

import java.util.ArrayList;
import java.util.List;

public class Fires extends Sprite {
    List<Spritable> fires = new ArrayList<>();

    public void render() {
        for (Spritable f : fires) {
            f.render();
        }
    }

    public void update() {
        for (Spritable f : fires) {
            f.update();
        }
    }

    public void fire(Spritable s) throws SlickException {
        fires.add(s);
    }
}

File: src/pewmode/Fires.kt
package pewmode

import org.newdawn.slick.Sound
import java.util.ArrayList

class Fires : SpriteGroup() {
    internal var fires: MutableList<Sprite> = ArrayList()
    val snd = Sound("sounds/lasers/1.wav")

    override fun render() {
        fires.forEach { it.render() }
    }

    override fun update() {
        fires.forEach { it.update() }

        if (fires.size > 100) {
            var temp : MutableList<Sprite> = ArrayList()
            fires.forEach { if(it.getxPos() < 3000) temp.add(it) }
            fires = temp
        }
    }

    internal fun fire(s: Sprite) {
        snd.play()
        fires.add(s)
    }

    override fun members(): MutableList<Sprite> {
        return fires
    }
}
